src/main/java/software/amazon/smithy/lsp/ext/SmithyProject.java

```diff
@@ -190,61 +194,157 @@ private static Map<ShapeId, Location> collectLocations(Model model) {
+                        Position endPosition = getEndPosition(memberEndMarker, lines);
+                        // Advance the member end marker on any non-mixin traits on the current member, so that the next
+                        // member location end is correct. Mixin traits will have been declared outside the
+                        // containing shape and shouldn't impact determining the end location of the next member.
+                        List<Trait> traits = memberShape.getAllTraits().values().stream()
+                                .filter(trait -> !trait.getSourceLocation().equals(SourceLocation.NONE))
+                                .filter(trait -> trait.getSourceLocation().getFilename().equals(modelFile))
+                                .filter(trait -> !isFromMixin(containerLocationRange, trait))
+                                .collect(Collectors.toList());
+                        if (!traits.isEmpty()) {
+                            traits.sort(new SourceLocationSorter());
+                            memberEndMarker = traits.get(0).getSourceLocation().getLine();
+                        }
+                        locations.put(memberShape.getId(), createLocation(modelFile, startPosition, endPosition));
+                        previousLine = currentLine;
+                    }
                 }
-                Location location =  new Location(getUri(modelFile), new Range(startPosition, endPosition));
-                locations.put(shape.getId(), location);
             }
         }
         return locations;
     }
 
+    // Use an empty range at the container's start since elided members are not present in the model file.
+    private static Location createElidedMemberLocation(Location containerLocation) {
+        Position startPosition = containerLocation.getRange().getStart();
+        Range memberRange = new Range(startPosition, startPosition);
+        return new Location(containerLocation.getUri(), memberRange);
+    }
+
+    // If the trait was defined outside the container, it was mixed in.
+    private static boolean isFromMixin(Range containerRange, Trait trait) {
+        int traitLocationLine = trait.getSourceLocation().getLine();
+        if (traitLocationLine < containerRange.getStart().getLine()
+                || traitLocationLine > containerRange.getEnd().getLine()) {
+            return true;
+        }
+        return false;
+    }
+
+    // Get the operation that matches an inlined input or output structure.
+    private static Optional<OperationShape> getOperationForInlinedInputOrOutput(Model model, Shape shape,
+                                                                                DocumentPreamble preamble) {
+        if (preamble.getIdlVersion().isPresent()) {
+            if (preamble.getIdlVersion().get().startsWith("2") && shape.isStructureShape()
+                    && (shape.hasTrait(OutputTrait.class) || shape.hasTrait(InputTrait.class))) {
```

Comment:
Not only inline input/output can have these traits, and they can have names that don't match the conventions.

Reply:
There was a slightly different issue here.

This was identifying inputs and outputs that matched the naming convention for inline inputs and outputs as having been inlined, when that was not necessarily the case.

I added a check in [0ed9978](https://github.com/awslabs/smithy-language-server/pull/61/commits/0ed9978ed9b3d58df77fb1af80310514f8d352bd) to be sure that input/output structures that matched the naming actually were defined inline with the `:=` operator.